utils/docker/Dockerfile.el.9

```diff
@@ -0,0 +1,206 @@
+# environment for building the DAOS project.
+#
+
+# Pull base image
+ARG BASE_DISTRO=almalinux:9
+FROM $BASE_DISTRO
+LABEL maintainer="daos@daos.groups.io"
+
+# Intermittent cache-bust.  Used to reduce load on the actual CACHEBUST later.
+ARG CB0
+
+# Use local repo server if present
+ARG REPO_FILE_URL
+RUN set -e;                                                      \
+    if [ -n "$REPO_FILE_URL" ]; then                             \
+        cd /etc/yum.repos.d/ &&                                  \
+        curl -k -f -o daos_ci-el9-artifactory.repo.tmp           \
+             "$REPO_FILE_URL"daos_ci-el9-artifactory.repo &&     \
+        for file in *.repo; do                                   \
+            true > $file;                                        \
+        done;                                                    \
+        mv daos_ci-el9-artifactory.repo{.tmp,};                  \
+    fi;                                                          \
+    dnf --assumeyes install epel-release dnf-plugins-core;       \
+    dnf config-manager --save --setopt=assumeyes=True;           \
+    dnf config-manager --save --setopt=install_weak_deps=False;  \
+    if [ -n "$REPO_FILE_URL" ]; then                             \
+        PT_REPO=daos_ci-rocky8-crb-artifactory;                  \
+    else                                                         \
+        PT_REPO=crb;                                             \
+    fi;                                                          \
+    dnf -y config-manager --enable $PT_REPO;                     \
+    dnf -y clean all
+
+ARG JENKINS_URL
+ARG REPOS
+RUN for repo in $REPOS; do                                                \
+        branch="master";                                                  \
+        build_number="lastSuccessfulBuild";                               \
+        if [[ $repo = *@* ]]; then                                        \
+            branch="${repo#*@}";                                          \
+            repo="${repo%@*}";                                            \
+            if [[ $branch = *:* ]]; then                                  \
+                build_number="${branch#*:}";                              \
+                branch="${branch%:*}";                                    \
+            fi;                                                           \
+        fi;                                                               \
+        echo -e "[$repo:$branch:$build_number]\n\
+name=$repo:$branch:$build_number\n\
+baseurl=${JENKINS_URL}job/daos-stack/job/$repo/job/$branch/$build_number/artifact/artifacts/el9/\n\
+enabled=1\n\
+gpgcheck=False\n" >> /etc/yum.repos.d/$repo:$branch:$build_number.repo;   \
+        cat /etc/yum.repos.d/$repo:$branch:$build_number.repo; \
+        dnf repolist; \
+        dnf --disablerepo=\* --enablerepo=$repo:$branch:$build_number makecache; \
```

Comment:
These two lines are removed in master as well, they install a cache of the repo contents which we go to great steps to avoid in other places, and reading this cache was also causing overlayfs problems.